[BUG] SQLite Errors when accessing 'Security' page

**Describe the bug**
After allowing docker container to update to latest  version (3.9.0) today I can no longer access the 'Security' page.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to meshmonitor homepage
2. Log in
3. Click on 'Security' page
4. See error "Error: Failed to get security issues" and/or "Error: Failed to fetch key mismatch history"

**Expected behavior**
The security page to load as normal

**Environment (please complete the following information):**
* Host OS: [Ubuntu Linux 24.04]
* Deployment type: Docker version 29.3.0 (meshmonitor version 3.9.0)
* Database backend: SQLite
* Node type: Station G2
* Node Connection Type: Wifi

**Additional context**
Docker log output:
```
[ERROR] Error getting security issues: SqliteError: no such column: "lastMeshReceivedKey" - should this be a string literal in single-quotes?
    at Database.prepare (/app/node_modules/better-sqlite3/lib/methods/wrappers.js:5:21)
    at BetterSQLiteSession.prepareQuery (file:///app/node_modules/drizzle-orm/better-sqlite3/session.js:23:30)
    at BetterSQLiteSession.prepareOneTimeQuery (file:///app/node_modules/drizzle-orm/sqlite-core/session.js:141:17)
    at SQLiteSelectBase._prepare (file:///app/node_modules/drizzle-orm/sqlite-core/query-builders/select.js:615:88)
    at SQLiteSelectBase.all (file:///app/node_modules/drizzle-orm/sqlite-core/query-builders/select.js:641:17)
    at SQLiteSelectBase.execute (file:///app/node_modules/drizzle-orm/sqlite-core/query-builders/select.js:650:17)
    at SQLiteSelectBase.then (file:///app/node_modules/drizzle-orm/query-promise.js:21:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:104:5) {
  code: 'SQLITE_ERROR'
}
[ERROR] Error fetching key mismatch history: SqliteError: no such column: oldKeyFragment
    at Database.prepare (/app/node_modules/better-sqlite3/lib/methods/wrappers.js:5:21)
    at DatabaseService.getKeyRepairLogAsync (file:///app/dist/services/database.js:6364:30)
    at file:///app/dist/server/routes/securityRoutes.js:264:43
    at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17)
    at next (/app/node_modules/router/lib/route.js:157:13)
    at Route.dispatch (/app/node_modules/router/lib/route.js:117:3)
    at handle (/app/node_modules/router/index.js:435:11)
    at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17)
    at /app/node_modules/router/index.js:295:15
    at processParams (/app/node_modules/router/index.js:582:12) {
  code: 'SQLITE_ERROR'
}
```

--- a/src/server/migrations/084_add_key_mismatch_columns.ts
+++ b/src/server/migrations/084_add_key_mismatch_columns.ts
@@ -10,18 +10,24 @@ export function runMigration084Sqlite(db: Database): void {
     db.exec("ALTER TABLE nodes ADD COLUMN lastMeshReceivedKey TEXT");
   }
 
-  const hasOldKeyFragment = db.prepare(
-    "SELECT COUNT(*) as count FROM pragma_table_info('auto_key_repair_log') WHERE name='oldKeyFragment'"
+  // auto_key_repair_log may not exist if user never enabled auto-key management (created in migration 046)
+  const hasRepairLogTable = db.prepare(
+    "SELECT COUNT(*) as count FROM sqlite_master WHERE type='table' AND name='auto_key_repair_log'"
   ).get() as { count: number };
-  if (hasOldKeyFragment.count === 0) {
-    db.exec("ALTER TABLE auto_key_repair_log ADD COLUMN oldKeyFragment TEXT");
-  }
+  if (hasRepairLogTable.count > 0) {
+    const hasOldKeyFragment = db.prepare(
+      "SELECT COUNT(*) as count FROM pragma_table_info('auto_key_repair_log') WHERE name='oldKeyFragment'"
+    ).get() as { count: number };
+    if (hasOldKeyFragment.count === 0) {
+      db.exec("ALTER TABLE auto_key_repair_log ADD COLUMN oldKeyFragment TEXT");
+    }
 
-  const hasNewKeyFragment = db.prepare(
-    "SELECT COUNT(*) as count FROM pragma_table_info('auto_key_repair_log') WHERE name='newKeyFragment'"
-  ).get() as { count: number };
-  if (hasNewKeyFragment.count === 0) {
-    db.exec("ALTER TABLE auto_key_repair_log ADD COLUMN newKeyFragment TEXT");
+    const hasNewKeyFragment = db.prepare(
+      "SELECT COUNT(*) as count FROM pragma_table_info('auto_key_repair_log') WHERE name='newKeyFragment'"
+    ).get() as { count: number };
+    if (hasNewKeyFragment.count === 0) {
+      db.exec("ALTER TABLE auto_key_repair_log ADD COLUMN newKeyFragment TEXT");
+    }
   }
 }
 
@@ -33,18 +39,24 @@ export async function runMigration084Postgres(client: PoolClient): Promise<void>
     await client.query('ALTER TABLE nodes ADD COLUMN "lastMeshReceivedKey" TEXT');
   }
 
-  const oldKeyCheck = await client.query(
-    "SELECT column_name FROM information_schema.columns WHERE table_name = 'auto_key_repair_log' AND column_name = 'oldKeyFragment'"
+  // auto_key_repair_log may not exist if user never enabled auto-key management (created in migration 046)
+  const hasRepairLogTable = await client.query(
+    "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public' AND table_name = 'auto_key_repair_log'"
   );
-  if (oldKeyCheck.rows.length === 0) {
-    await client.query('ALTER TABLE auto_key_repair_log ADD COLUMN "oldKeyFragment" VARCHAR(8)');
-  }
+  if (hasRepairLogTable.rows.length > 0) {
+    const oldKeyCheck = await client.query(
+      "SELECT column_name FROM information_schema.columns WHERE table_name = 'auto_key_repair_log' AND column_name = 'oldKeyFragment'"
+    );
+    if (oldKeyCheck.rows.length === 0) {
+      await client.query('ALTER TABLE auto_key_repair_log ADD COLUMN "oldKeyFragment" VARCHAR(8)');
+    }
 
-  const newKeyCheck = await client.query(
-    "SELECT column_name FROM information_schema.columns WHERE table_name = 'auto_key_repair_log' AND column_name = 'newKeyFragment'"
-  );
-  if (newKeyCheck.rows.length === 0) {
-    await client.query('ALTER TABLE auto_key_repair_log ADD COLUMN "newKeyFragment" VARCHAR(8)');
+    const newKeyCheck = await client.query(
+      "SELECT column_name FROM information_schema.columns WHERE table_name = 'auto_key_repair_log' AND column_name = 'newKeyFragment'"
+    );
+    if (newKeyCheck.rows.length === 0) {
+      await client.query('ALTER TABLE auto_key_repair_log ADD COLUMN "newKeyFragment" VARCHAR(8)');
+    }
   }
 }
 
@@ -56,17 +68,24 @@ export async function runMigration084Mysql(pool: Pool): Promise<void> {
     await pool.query('ALTER TABLE nodes ADD COLUMN lastMeshReceivedKey VARCHAR(128)');
   }
 
-  const [oldKeyRows] = await pool.query(
-    "SELECT COLUMN_NAME FROM information_schema.columns WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log' AND column_name = 'oldKeyFragment'"
+  // auto_key_repair_log may not exist if user never enabled auto-key management (created in migration 046)
+  const [repairLogTable] = await pool.query(
+    "SELECT TABLE_NAME FROM information_schema.tables WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log'"
   );
-  if ((oldKeyRows as any[]).length === 0) {
-    await pool.query('ALTER TABLE auto_key_repair_log ADD COLUMN oldKeyFragment VARCHAR(8)');
-  }
+  if ((repairLogTable as any[]).length > 0) {
+    const [oldKeyRows] = await pool.query(
+      "SELECT COLUMN_NAME FROM information_schema.columns WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log' AND column_name = 'oldKeyFragment'"
+    );
+    if ((oldKeyRows as any[]).length === 0) {
+      await pool.query('ALTER TABLE auto_key_repair_log ADD COLUMN oldKeyFragment VARCHAR(8)');
+    }
 
-  const [newKeyRows] = await pool.query(
-    "SELECT COLUMN_NAME FROM information_schema.columns WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log' AND column_name = 'newKeyFragment'"
-  );
-  if ((newKeyRows as any[]).length === 0) {
-    await pool.query('ALTER TABLE auto_key_repair_log ADD COLUMN newKeyFragment VARCHAR(8)');
+    const [newKeyRows] = await pool.query(
+      "SELECT COLUMN_NAME FROM information_schema.columns WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log'"
+      + " AND column_name = 'newKeyFragment'"
+    );
+    if ((newKeyRows as any[]).length === 0) {
+      await pool.query('ALTER TABLE auto_key_repair_log ADD COLUMN newKeyFragment VARCHAR(8)');
+    }
   }
 }

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -7649,9 +7649,24 @@ class DatabaseService {
     if (this.drizzleDbType === 'postgres') {
       const client = await (this as any).pgPool.connect();
       try {
+        // Check if table exists (may not exist if auto-key management was never enabled)
+        const tableCheck = await client.query(
+          "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public' AND table_name = 'auto_key_repair_log'"
+        );
+        if (tableCheck.rows.length === 0) {
+          return [];
+        }
+
+        // Check if migration 084 columns exist
+        const colCheck = await client.query(
+          "SELECT column_name FROM information_schema.columns WHERE table_name = 'auto_key_repair_log' AND column_name = 'oldKeyFragment'"
+        );
+        const selectCols = colCheck.rows.length > 0
+          ? 'id, timestamp, "nodeNum", "nodeName", action, success, "oldKeyFragment", "newKeyFragment"'
+          : 'id, timestamp, "nodeNum", "nodeName", action, success';
+
         const result = await client.query(
-          `SELECT id, timestamp, "nodeNum", "nodeName", action, success, "oldKeyFragment", "newKeyFragment"
-           FROM auto_key_repair_log ORDER BY timestamp DESC LIMIT $1`,
+          `SELECT ${selectCols} FROM auto_key_repair_log ORDER BY timestamp DESC LIMIT $1`,
           [limit]
         );
         return result.rows.map((row: any) => ({
@@ -7669,9 +7684,25 @@ class DatabaseService {
       }
     } else if (this.drizzleDbType === 'mysql') {
       const pool = (this as any).mysqlPool;
+
+      // Check if table exists (may not exist if auto-key management was never enabled)
+      const [tableRows] = await pool.query(
+        "SELECT TABLE_NAME FROM information_schema.tables WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log'"
+      );
+      if ((tableRows as any[]).length === 0) {
+        return [];
+      }
+
+      // Check if migration 084 columns exist
+      const [colRows] = await pool.query(
+        "SELECT COLUMN_NAME FROM information_schema.columns WHERE table_schema = DATABASE() AND table_name = 'auto_key_repair_log' AND column_name = 'oldKeyFragment'"
+      );
+      const selectCols = (colRows as any[]).length > 0
+        ? 'id, timestamp, nodeNum, nodeName, action, success, oldKeyFragment, newKeyFragment'
+        : 'id, timestamp, nodeNum, nodeName, action, success';
+
       const [rows] = await pool.query(
-        `SELECT id, timestamp, nodeNum, nodeName, action, success, oldKeyFragment, newKeyFragment
-         FROM auto_key_repair_log ORDER BY timestamp DESC LIMIT ?`,
+        `SELECT ${selectCols} FROM auto_key_repair_log ORDER BY timestamp DESC LIMIT ?`,
         [limit]
       );
       return (rows as any[]).map((row: any) => ({
@@ -7685,9 +7716,24 @@ class DatabaseService {
         newKeyFragment: row.newKeyFragment || null,
       }));
     }
-    // SQLite — query directly with new columns (available after migration 084)
+    // SQLite — check if table exists first (may not exist if auto-key management was never enabled)
+    const hasTable = this.db.prepare(
+      "SELECT COUNT(*) as count FROM sqlite_master WHERE type='table' AND name='auto_key_repair_log'"
+    ).get() as { count: number };
+    if (hasTable.count === 0) {
+      return [];
+    }
+
+    // Check if migration 084 columns exist
+    const hasOldKeyCol = this.db.prepare(
+      "SELECT COUNT(*) as count FROM pragma_table_info('auto_key_repair_log') WHERE name='oldKeyFragment'"
+    ).get() as { count: number };
+    const selectCols = hasOldKeyCol.count > 0
+      ? 'id, timestamp, nodeNum, nodeName, action, success, oldKeyFragment, newKeyFragment'
+      : 'id, timestamp, nodeNum, nodeName, action, success';
+
     const rows = this.db.prepare(`
-      SELECT id, timestamp, nodeNum, nodeName, action, success, oldKeyFragment, newKeyFragment
+      SELECT ${selectCols}
       FROM auto_key_repair_log ORDER BY timestamp DESC LIMIT ?
     `).all(limit) as any[];
     return rows.map((row: any) => ({

